stan-ballerina/Module.md

```diff
@@ -0,0 +1,131 @@
+```
+
+3. Connect to a server with custom configurations:
+```ballerina
+stan:StreamingConfiguration config = {
+  clusterId: "test-cluster",
+  ackTimeout: 30,
+  connectionTimeout: 5;
+};
+stan:Client stanClient = check new("nats://localhost:4222",  config);
+```
+
+#### Publishing Messages
+
+Once connected, use the `publishMessage` function to publish messages to a given subject as shown below.
+
+```ballerina
+string message = "hello world";
+check producer->publishMessage({ content: message, subject: "demo" });
+```
+
+#### Listening to incoming messages
+
+```ballerina
+// Binds the consumer to listen to the messages published to the 'demo' subject.
+@stan:ServiceConfig {
+    subject: "demo"
+}
+service stan:Service on subscription {
+    
+    remote function onMessage(stan:Message message) {
+    }
+}
+```
+### Advanced Usage
+
+#### Setting up TLS
+
+The Ballerina NATS streaming module allows the use of TLS in communication. This setting expects a secure socket to be
+set in the connection configuration as shown below.
+
+##### Configuring TLS in the `stan:Listener`
+```ballerina
+stan:SecureSocket secured = {
+    cert: {
+        path: "<path>/truststore.p12",
+        password: "password"
+    },
+    key: {
+        path: "<path>/keystore.p12",
+        password: "password"
+    }
+};
+stan:Listener stanListener = check new("nats://serverone:4222", secureSocket = secured);
+```
+
+##### Configuring TLS in the `stan:Client`
+```ballerina
+stan:SecureSocket secured = {
+    cert: {
+        path: "<path>/truststore.p12",
+        password: "password"
+    },
+    key: {
+        path: "<path>/keystore.p12",
+        password: "password"
+    }
+};
+stan:Client stanClient = check new("nats://serverone:4222", secureSocket = secured);
+```
+
+#### Acknowledging Messages
+
+NATS Streaming offers At-Least-Once delivery semantics, meaning that once a message has been delivered to an eligible subscriber, if an acknowledgement is not received within the configured timeout interval, NATS Streaming will attempt redelivery of the message.
+If you need to acknowledge the incoming messages manually, make sure to set the `autoAck` status of the service config to false as shown below.
+
+```ballerina
+// Set `autoAck` to false.
+@stan:ServiceConfig {
+    subject: "demo",
+    autoAck: false
+}
+service stan:Service on subscription {
+    
+    remote function onMessage(stan:Message message, stan:Caller caller) {
+        // Manually acknowledge the message received.
+        stan:Error? ackResult = caller->ack();
+    }
+}
+```
+
+#### Durable Subscriptions
+
+Durable subscriptions allow clients to assign a durable name to a subscription when it is created. Doing this causes the NATS Streaming server to track the last acknowledged message for that clientID + durable name, so that only messages since the last acknowledged message will be delivered to the client. These subscriptions will survive a server restart.
```

Comment:
```suggestion
Durable subscriptions allow clients to assign a durable name to a subscription when it is created. Doing this causes the NATS Streaming server to track the last-acknowledged message for that `clientID` + `durable name` so that only messages since the last acknowledged message will be delivered to the client. These subscriptions will survive a server restart.
```